source/shared/version.h

```diff
@@ -24,13 +24,24 @@
 
 #define SQLVERSION_MAJOR 4
 #define SQLVERSION_MINOR 1
-#define SQLVERSION_RELEASE 6
+#define SQLVERSION_PATCH 7
 #define SQLVERSION_BUILD 0
 
-#define VER_FILEVERSION_STR     STRINGIFY( SQLVERSION_MAJOR ) "." STRINGIFY( SQLVERSION_MINOR ) "." STRINGIFY( SQLVERSION_RELEASE ) "." STRINGIFY( SQLVERSION_BUILD ) 
-#define _FILEVERSION            SQLVERSION_MAJOR,SQLVERSION_MINOR,SQLVERSION_RELEASE,SQLVERSION_BUILD
-#define PHP_SQLSRV_VERSION      STRINGIFY( SQLVERSION_MAJOR ) "." STRINGIFY( SQLVERSION_MINOR ) "." STRINGIFY( SQLVERSION_RELEASE )
-#define PHP_PDO_SQLSRV_VERSION  PHP_SQLSRV_VERSION
+// Semantic versioning pre-release and buil metadata
+#define SEMVER_PRERELEASE "-preview"
+#define SEMVER_BUILDMETA
+
+#if SQLVERSION_BUILD > 0
+#undef SEMVER_BUILDMETA
+#define SEMVER_BUILDMETA "+" STRINGIFY( SQLVERSION_BUILD )
+#endif
+
+#define VER_FILEVERSION_STR     STRINGIFY( SQLVERSION_MAJOR ) "." STRINGIFY( SQLVERSION_MINOR ) "." STRINGIFY( SQLVERSION_PATCH ) SEMVER_PRERELEASE SEMVER_BUILDMETA
+#define _FILEVERSION            SQLVERSION_MAJOR,SQLVERSION_MINOR,SQLVERSION_PATCH,SQLVERSION_BUILD
+
```

Comment:
that's true and when it is 0, `VER_FILEVERSION_STR`  is `4.1.7-preview`, and build number is not counted in precedence order.